infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryDao.kt

```diff
@@ -1141,6 +1143,61 @@ class GeometryDao @Autowired constructor(
         }
     }
 
+
+
+
+    fun removeReferencesToTrackNumber(id: IntId<TrackLayoutTrackNumber>): RemovedTrackNumberReferenceIds {
+        val kmPostIds = removeKmPostReferenceToTrackNumber(id)
+        val alignmentIds = removeAlignmentReferenceToTrackNumber(id)
+        val planIds = removePlanReferenceToTrackNumber(id)
+
+        return RemovedTrackNumberReferenceIds(
+            kmPostIds = kmPostIds,
+            alignmentIds = alignmentIds,
+            planIds = planIds
+        )
+    }
+
+    private fun removeKmPostReferenceToTrackNumber(id: IntId<TrackLayoutTrackNumber>): List<IntId<GeometryKmPost>> {
+        val sql = """
+          update geometry.km_post
+          set track_number_id = null 
+          where track_number_id = :id
+          returning id as km_post_id
+        """.trimIndent()
+
+        return jdbcTemplate.query(sql, mapOf("id" to id.intValue)) { rs, _ ->
```

Comment:
Lisää DAO-tason logitukset näille alafunkkarille (ylätasolle ei tarvitse koska kutsuu vain muita). Esim näin:
```
return jdbcTemplate
  .query(sql, mapOf("id" to id.intValue)) { rs, _ -> IntId(rs.getInt("km_post_id")) }
  .also { ids -> logger.daoAccess(AccessType.UPDATE, GeometryKmPost::class, ids) }
```
... ja tuolta sisältä siis tuo GeometryKmPost::class aina päivitettävän rivin tyypin mukaan.